plugin/trino-jmx/src/test/java/io/trino/plugin/jmx/TestJmxHistoricalData.java

```diff
@@ -74,4 +80,23 @@ public void testWildCardPatterns()
 
         assertEquals(jmxHistoricalData.getTables(), ImmutableSet.of("java.lang:type=classloading", "java.lang:type=compilation"));
     }
+
+    @Test
+    public void testLoadDynamicTable() throws Exception
+    {
+        MBeanServer mbeanServer = getPlatformMBeanServer();
+        JmxHistoricalData jmxHistoricalData = new JmxHistoricalData(MAX_ENTRIES, ImmutableSet.of(TABLE_NAME, ON_FLY_TABLE_NAME), mbeanServer);
+
+        List<Integer> columns = ImmutableList.of(0);
+        assertEquals(jmxHistoricalData.getRows(TABLE_NAME, columns), ImmutableList.of());
+
+        StandardMBean newMBean = new StandardMBean(new ArrayList<>(), List.class);
+        mbeanServer.registerMBean(newMBean, ObjectName.getInstance(ON_FLY_TABLE_NAME));
```

Comment:
Instead of injecting a new Bean can we try accessing the `InternalResourceGroup` - mentioned in the actual issue ?